ReactiveCocoa/Objective-C/RACSignal.m

```diff
@@ -107,31 +107,36 @@ - (RACSignal *)bind:(RACStreamBindBlock (^)(void))block {
 	return [[RACSignal createSignal:^(id<RACSubscriber> subscriber) {
 		RACStreamBindBlock bindingBlock = block();
 
-		NSMutableArray *signals = [NSMutableArray arrayWithObject:self];
+		__block NSUInteger signalsCount = 1;   // indicates self
```

Comment:
Actually, this specific issue is solvable by checking the return value of `OSAtomicDecrement32Barrier`. However, it seems that if `signalsCount` reaches zero you must `dispose` the disposable and send completion to the subscriber, so you must put this entire logic in a critical section, as you've already did.